libs/barista-components/stacked-series-chart/src/stacked-series-chart.ts

```diff
@@ -292,7 +293,12 @@ export class DtStackedSeriesChart implements OnDestroy {
     node?: DtStackedSeriesChartNode,
   ): void {
     if (this._selectable) {
-      if (series && node && this._selected[1] !== node) {
+      if (
+        series &&
+        (!this._selected[0] ||
+          (this._selected[0] && series.label !== this._selected[0].label) ||
+          this._selected[1] !== node)
+      ) {
```

Comment:
Please add a comment that explains your check as it is really monstrous and hard to read